controllers/medusa/medusabackupjob_controller_test.go

```diff
@@ -488,6 +525,26 @@ func findDatacenterCondition(status *cassdcapi.CassandraDatacenterStatus, condTy
 	return nil
 }
 
+func deleteDatacenterPods(t *testing.T, f *framework.Framework, ctx context.Context, dcKey framework.ClusterKey, dc *cassdcapi.CassandraDatacenter) {
+	for i := int32(0); i < dc.Spec.Size; i++ {
+		pod := &corev1.Pod{}
+		podName := fmt.Sprintf("%s-%s-%d", dc.Spec.ClusterName, dc.DatacenterName(), i)
+		podKey := framework.NewClusterKey(dcKey.K8sContext, dcKey.Namespace, podName)
+		err := f.Get(ctx, podKey, pod)
```

Comment:
Why do you need a get? Just do Delete().

Reply:
Although I'm not sure why you're not just running on another namespace this test and just don't generate fake pods there. Seems easier.

Reply:
I've seen this pattern somewhere else. In the rest api, I think, when answering a DELETE (of cql or dataapi connectivity), we `get` before `delete`.
One thing this allows is to graciously error out if the object we don't delete does not exists. But, that's not what we always want.
In this case, I'd stick how it's done, but I'll gladly switch to doing this differently in the future.

Reply:
Well, the issue is that your Get is redundant or can give you wrong answer. If the object does not exists, what you get is:

Already deleted:

Get -> 404 (errors.IsNotFound())
Or:
Delete -> 404 (errors.IsNotFound())

Alternatively, since Get is cached, you might get:

Get -> 200
Delete -> 404 (errors.IsNotFound())

Existing:

Get -> 200
Delete -> 200
Or
Delete -> 200

In theory, following is also possible:

Get -> 404 (errors.IsNotFound())
(whoops, item is actually there)
Or:
Delete -> 200

The Get does not make anything better. If it's done somewhere else, it's a mistake. What it can make is give you wrong answer only for the Delete command, since Get command is eventually consistent (it comes from local cache), while Delete is consistent command.

But it will never make anything better.

Reply:
Re-read this again. Makes sense. Pushed a commit removing the get.